src/python/turicreate/toolkits/image_analysis/image_analysis.py

```diff
@@ -177,3 +190,109 @@ def resize(image, width, height, channels=None, decode=False, resample="nearest"
+                                          Exported Core ML model will be ~41K.
+
+        Models are downloaded from the internet if not available locally. Once
+        downloaded, the models are cached for future use.
+
+    Returns
+    -------
+    out : SArray
+        Returns an SArray with all the extracted features.
+
+    Examples
+    --------
+
+    """
+
+    # Check model parameter
+    allowed_models = list(_pre_trained_models.IMAGE_MODELS.keys())
+    if _mac_ver() >= (10, 14):
+        allowed_models.append("VisionFeaturePrint_Scene")
+    _tkutl._check_categorical_option_type("model", model_name, allowed_models)
+
+    # Check images parameter
+    if not isinstance(images, _tc.SArray):
+        raise TypeError("Unrecognized type for 'images'. An SArray is expected.")
+    if len(images) == 0:
+        raise _ToolkitError("Unable to extract features on an empty SArray object")
+
+    if batch_size < 1:
+        raise ValueError("'batch_size' must be greater than or equal to 1")
+
+    # Extract features
+    feature_extractor = _image_feature_extractor._create_feature_extractor(model_name)
+    images_sf = _tc.SFrame({"image":images})
+    return feature_extractor.extract_features(images_sf, "image", verbose=verbose, 
+        batch_size=batch_size)
+
+
+def find_only_image_extracted_features_column(sframe, model_name):
+    """
+    Finds the only column in `sframe` with a type of array.array and has
+    the length same as the last layer of the model in use.
+    If there are zero or more than one image columns, an exception will
+    be raised.
+    """
+    from array import array
+
+    try:
+        feature_columns, _ = zip(*list(filter(lambda x: x[1] == array, list(zip(sframe.column_names(), sframe.column_types())))))
+    except ValueError:
+        raise _ToolkitError(
+            'No "{col_name}" column specified and no column with expected type "{type_name}" is found.'.format(
+                col_name="extracted_features", type_name="array"
+            )
+            + ' "datasets" consists of columns with types: '
+            + ", ".join([x.__name__ for x in sframe.column_types()])
+            + "."
+        )
+
+    feature_columns = list(filter(lambda x: is_image_deep_feature_sarray(sframe[x], model_name), feature_columns))
+    if len(feature_columns) == 1:
+        return feature_columns[0]
+    elif len(feature_columns) > 1:
+        raise _ToolkitError(
+            'No "{col_name}" column specified and more than one extracted features {type_name} column in "dataset". Can not infer correct {col_name} column.'.format(
+                col_name="extracted_features", type_name="array"
+            )
+        )
+
+
+def is_image_deep_feature_sarray(feature_sarray, model_name):
+    """
+    Finds if the given `SArray` has extracted features for a given model_name.
+    """
+    from array import array
+
+    if feature_sarray.dtype != array:
+        return False
+    if type(feature_sarray[0]) != array:
```

Comment:
Before executing this line, you should check `len(feature_sarray) > 0`.

Reply:
You are right, I should have checked for the `feature_sarray` length. I will update this in my next PR